### ISSUE TITLE
TypeError: node.children is not iterable

### ISSUE THREAD

**Issue by OneCyrus** [warrenseine/prettier-plugin-csharp#6]
for the following code i get an error

```
using MyProject.SelfService.Logic.APIHelper;
using GraphQL.Language.AST;
using GraphQL.Types;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.Web;

namespace MyProject.AuthPortal.GraphQL.Customer.Service.SecurityPatching
{
    public class SecurityPatchingModel
    {

        private readonly CMDBApiHelper<SecurityPatchingModel> _cmdbHelper;
        private readonly String Customer;

        public SecurityPatchingModel(UserContext userContext, String customer)
        {
            Customer = customer;
            _cmdbHelper = new CMDBApiHelper<SecurityPatchingModel>(userContext.GetAccessToken());
        }

        public async Task<List<API.CMDB.Schema.Server.Server>> GetServers<T>(ResolveFieldContext<T> context, string[] skipPaths = null)
        {
            var arguments = new Arguments();
            var whereFilterArgument = new Argument(new NameNode("where"));
            {
                var fields = new List<ObjectField>(new ObjectField[] {
                    new ObjectField(
                        new NameNode("customer"),
                        new StringValue(Customer.ToLower())
                        ),
                    new ObjectField(
                        new NameNode("active"),
                        new EnumValue("ACTIVE")
                        ),
                    });
                
                whereFilterArgument.Value = new ObjectValue(fields);
            }
            arguments.Add(whereFilterArgument);
            if (
                context.Arguments.ContainsKey("where") &&
                context.Arguments["where"] is Dictionary<string, object> where &&
                where.ContainsKey("uid") && where["uid"] is string uid && whereFilterArgument.Value is ObjectValue objVal
                )
            {

                whereFilterArgument.Value = new ObjectValue(objVal.ObjectFields.Append(new ObjectField(new NameNode("uid"), new StringValue(uid))));
            }

            return await _cmdbHelper.QueryDepthOne<List<API.CMDB.Schema.Server.Server>, T>(context, fieldName: "servers", arguments: arguments, skipPaths: skipPaths);
        }
    }
}
```

```
GraphQL\Customer\Service\SecurityPatching\SecurityPatchingModel.csline 45:73 missing ')' at 'where'
line 46:67 missing ';' at 'uid'
line 46:115 missing ';' at 'objVal'

[error] GraphQL\Customer\Service\SecurityPatching\SecurityPatchingModel.cs: TypeError: node.children is not iterable
[error]     at simplifyTree (C:\Source\MyProject.SelfService\src\MyProject.SelfService.Portal\node_modules\prettier-csharp\src\parser.js:69:26)
[error]     at simplifyTree (C:\Source\MyProject.SelfService\src\MyProject.SelfService.Portal\node_modules\prettier-csharp\src\parser.js:70:13)
[error]     at simplifyTree (C:\Source\MyProject.SelfService\src\MyProject.SelfService.Portal\node_modules\prettier-csharp\src\parser.js:70:13)
[error]     at simplifyTree (C:\Source\MyProject.SelfService\src\MyProject.SelfService.Portal\node_modules\prettier-csharp\src\parser.js:70:13)
[error]     at simplifyTree (C:\Source\MyProject.SelfService\src\MyProject.SelfService.Portal\node_modules\prettier-csharp\src\parser.js:70:13)
[error]     at simplifyTree (C:\Source\MyProject.SelfService\src\MyProject.SelfService.Portal\node_modules\prettier-csharp\src\parser.js:70:13)
[error]     at simplifyTree (C:\Source\MyProject.SelfService\src\MyProject.SelfService.Portal\node_modules\prettier-csharp\src\parser.js:70:13)
[error]     at simplifyTree (C:\Source\MyProject.SelfService\src\MyProject.SelfService.Portal\node_modules\prettier-csharp\src\parser.js:70:13)
[error]     at simplifyTree (C:\Source\MyProject.SelfService\src\MyProject.SelfService.Portal\node_modules\prettier-csharp\src\parser.js:70:13)
[error]     at simplifyTree (C:\Source\MyProject.SelfService\src\MyProject.SelfService.Portal\node_modules\prettier-csharp\src\parser.js:70:13)
```

**Comment by warrenseine** [comment]
Indeed, you are using a C# 7.0 feature called "Is-expressions with patterns", while the grammar currently supports C# 6 only.

I will try to make it work.

**Comment by warrenseine** [comment]
Fixed with 0.4.0-development.

**Comment by warrenseine** [comment]
Additionally, I'm working on getting cleaner error messages when parsing fails. Lots of C# 7 features are not supported yet. See below a failure with tuples:

![cleaner_error_message](https://user-images.githubusercontent.com/539804/51382909-4b6b5080-1b18-11e9-9b8c-d4733c3f99b2.png)
